hal/armv7a/cpu.c

```diff
@@ -87,21 +152,105 @@ int hal_cpuCreateContext(cpu_context_t **nctx, void *start, void *kstack, size_t
 }
 
 
+void hal_cpuRestore(cpu_context_t *curr, cpu_context_t *next)
+{
+	curr->savesp = (u32)next /*+ sizeof(u32)*/;
+}
+
+
+void hal_cpuSetReturnValue(cpu_context_t *ctx, int retval)
+{
+	ctx->r0 = retval;
+}
+
+
+u32 hal_cpuGetPC(void)
+{
+	void *pc;
+
+	__asm__ volatile ("mov %0, pc" : "=r" (pc));
+
+	return (u32)pc;
+}
+
+
+void *hal_cpuGetSP(cpu_context_t *ctx)
+{
+	return (void *)ctx;
+}
+
+
+void *hal_cpuGetUserSP(cpu_context_t *ctx)
+{
+	return (void *)ctx->sp;
+}
+
+
+int hal_cpuSupervisorMode(cpu_context_t *ctx)
+{
+	return ctx->psr & 0xf;
+}
+
+
+int hal_cpuPushSignal(void *kstack, void (*handler)(void), int n)
+{
+	cpu_context_t *ctx = (void *)((char *)kstack - sizeof(cpu_context_t));
+
+	/* No signal handling inside IT block */
+	if (ctx->psr & 0x600fc00)
+		return -1;
+
+	PUTONSTACK(ctx->sp, u32, ctx->pc | !!(ctx->psr & THUMB_STATE));
+	PUTONSTACK(ctx->sp, int, n);
+
+	ctx->pc = (u32)handler & ~1;
+
+	if ((u32)handler & 1)
+		ctx->psr |= THUMB_STATE;
+	else
+		ctx->psr &= ~THUMB_STATE;
+
+	return 0;
+}
+
+
+void hal_cpuDataMemoryBarrier(void)
+{
+	__asm__ volatile ("dmb");
+}
+
+
+void hal_cpuDataSyncBarrier(void)
+{
+	__asm__ volatile ("dsb");
+}
+
+
+void hal_cpuInstrBarrier(void)
+{
+	__asm__ volatile ("isb");
+}
+
+
+unsigned int hal_cpuGetID(void)
+{
+	return 0;
+}
+
+
+unsigned int hal_cpuGetCount(void)
+{
+	return 1;
+}
+
+
 char *hal_cpuInfo(char *info)
 {
 	size_t n = 0;
 	u32 midr;
 
-#ifdef CPU_IMX
-	hal_strcpy(&info[n], "i.MX 6ULL ");
-	n += 10;
-#elif CPU_IMX6UL
-	hal_strcpy(&info[n], "i.MX 6UL ");
-	n += 9;
-#else
-	hal_strcpy(&info[n], "unknown ");
-	n += 8;
-#endif
+	hal_strcpy(&info[n], HAL_NAME_PLATFORM);
+	n += hal_strlen(HAL_NAME_PLATFORM);
```

Comment:
```suggestion
	hal_strcpy(info, HAL_NAME_PLATFORM);
	n = sizeof(HAL_NAME_PLATFORM) - 1;
```
The length of HAL_NAME_PLATFORM is known at compile time.
Maybe a better solution would be to use a pointer instead of an array in this function.

Reply:
Good suggestion, done.